packages/kbn-server-http-tools/src/get_listener.ts

```diff
@@ -52,3 +56,23 @@ const configureHttp1Listener = (
 
   return listener;
 };
+
+const configureHttp2Listener = (
```

Comment:
Yes, thanksfully node's listener interfaces are well done enough so that strictly technically speaking, just providing an `http2` listener instead of the expected `http` or `https` one to the HAPI server  "just works".

There are still a few minor issues caused by the fact that HAPI was never updated to officially support http2, e.g.s https://github.com/elastic/kibana/blob/69f81f24cc5351389c9c01835e2b5e096ff62ece/src/setup_node_env/exit_on_warning.js#L63-L69

But overall, it works out of the box (which was a great news because the cost/effort wouldn't have been the same if we were forced to fork or switch to another library 🙈)